p2p/src/config.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serde::{Deserialize, Serialize};
+
+use common::{chain::config::ChainType, primitives::semver::SemVer};
+
+/// The p2p subsystem configuration.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Config {
+    /// Address to bind P2P to.
+    pub address: String,
+    pub version: SemVer,
+    pub magic_bytes: [u8; 4],
```

Comment:
This is still a work in progress so feel free to ignore this comment but I'm not sure if this is correct. I believe the idea is to have `SemVer` as some kind of constant so user cannot specify it on the command line and same goes for the magic bytes, I don't think it's supposed to be controlled via command line.

Even if they were configurable, I'd still think chainstate configuration would be the right place and not P2P.

Reply:
Thank you for pointing this out! I'm almost sure you are right. I think I was too eager to move everything in the configuration.